AsyncUsageAnalyzers/AsyncUsageAnalyzers/Naming/AvoidAsyncSuffixAnalyzer.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace AsyncUsageAnalyzers.Naming
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    /// <summary>
+    /// This analyzer identifies methods which are not asynchronous according to the Task-based Asynchronous Pattern
+    /// (TAP) by their signature, and reports a warning if the method name includes the suffix <c>Async</c>.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AvoidAsyncSuffixAnalyzer : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="AvoidAsyncSuffixAnalyzer"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "AvoidAsyncSuffix";
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(NamingResources.AvoidAsyncSuffixTitle), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(NamingResources.AvoidAsyncSuffixMessageFormat), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly string Category = "AsyncUsage.CSharp.Naming";
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(NamingResources.AvoidAsyncSuffixDescription), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly string HelpLink = null;
```

Comment:
You could write a help page on the wiki part of this project and link to it here.